library/geo.py

```diff
@@ -55,3 +70,25 @@ def get_geo_data(place: str) -> Optional[Dict[str, str]]:
                         return {"place": res_place, "lat": lat, "lon": lon}
 
     return None
+
+
+def get_gsi_geo_data(place: str) -> Optional[Dict[str, str]]:
+    """
+    地名から座標を取得する(国土地理院版)
+    場所名が部分一致しているものを見つけて返すことで確度を上げている
+    :param place: 地名・住所
+    :return: place: 地名, lat: 緯度, lon: 経度
+    """
+
+    res = requests.get(
+        "https://msearch.gsi.go.jp/address-search/AddressSearch", {"q": place}
+    )
+
+    if res.status_code == 200:
```

Comment:
```suggestion
    if res.status_code != 200:
        return None
```

早期returnしてネストを浅くしても良いかも (suggestionは多分インデントが崩れているので別途コミットしてもらえると)。